matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -1371,6 +1376,54 @@ internal class DefaultCryptoService @Inject constructor(
         }
     }
 
+    override fun sendSharedHistoryKeysToLastChunk(roomId: String, userId: String, sessionInfoSet: Set<SessionInfoPair>?) {
+        cryptoCoroutineScope.launch(coroutineDispatchers.crypto) {
+            runCatching {
+                deviceListManager.downloadKeys(listOf(userId), false)
+            }.mapCatching {
+                val userDevices = cryptoStore.getUserDevices(userId)
+                userDevices?.forEach {
+                    // Lets share the provided inbound sessions for every user device
```

Comment:
general comment for the comments, they look like they could be replaced with functions

```kotlin
// Get inbound session from sessionId and sessionKey
sessionInfoSet?.inboundSessions()

// Filter only sessions with sharedHistory enabled
?.onlySessionsWithSharedHistory()

// Share the session to userId with deviceId
fun shareSessionWithUserDevice(session, roomId, userId, deviceId)
```